src/Chapter2.hs

```diff
@@ -841,7 +874,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate num list
+  | num < 0 = []
+  | otherwise = let len = (length list) + num in drop num (take len (cycle list))
```

Comment:
Unfortunately, `cycle` fails in runtime on empty lists :recycle: So you need to handle this case separately for this function to work properly.

Also, if the given number `num` is big, your function will be slow, since it will traverse the cyclic list for a very long time. Maybe you can think of a way to optimize this?